src/App.tsx

```diff
@@ -1,14 +1,58 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+import { getTodos } from './api';
+
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { Todo } from './types/Todo';
+
+const getFilteredTodos = (
+  param: string,
+  query: string,
+  todos: Todo[],
+) => {
+  let filteredTodos = [...todos];
+
+  if (param === 'active') {
+    filteredTodos = filteredTodos.filter(todo => !todo.completed);
+  } else if (param === 'completed') {
+    filteredTodos = filteredTodos.filter(todo => todo.completed);
+  }
+
+  if (query) {
+    filteredTodos = filteredTodos
+      .filter(todo => todo.title.toLocaleLowerCase()
+        .includes(query.toLocaleLowerCase()));
+  }
+
+  return filteredTodos;
+};
+
 export const App: React.FC = () => {
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filter, setFilter] = useState('All');
```

Comment:
You can create enum for filter params and use enum values for typification to avoid possible errors